portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -0,0 +1,68 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/home")
+public class HomeServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+    PrintWriter out = response.getWriter();
+    UserService userService = UserServiceFactory.getUserService();
+
+    // If user is not logged in, show a login form (could also redirect to a login page)
+    if (!userService.isUserLoggedIn()) {
+      String loginUrl = userService.createLoginURL("/");
+      out.println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
+      return;
+    }
+
+    // If user has not set a nickname, redirect to nickname page
+    String nickname = getUserNickname(userService.getCurrentUser().getUserId());
+    
+    if (nickname == null) {
+      response.sendRedirect("/nickname");
+      return;
+    }
+
+    // User is logged in and has a nickname, so the request can proceed
+    String logoutUrl = userService.createLogoutURL("/");
+    
+    out.println("<h1>Home</h1>");
+    out.println("<p>Hello " + nickname + "!</p>");
+    out.println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    out.println("<p>Change your nickname <a href=\"/nickname\">here</a>.</p>");
+    out.println("<form id=\"comment-input\"action=\"data\"method=\"POST\">");
+    out.println("<div class=\"form-group\">");
+    out.println("<p>Enter Comment</p>");   
+    out.println("<textarea class=\"form-control\" name=\"comment-input\" placeholder=\"Your Comment\"></textarea>");
+    out.println("</div>");
+    out.println("<button type=\"submit\" class=\"btn btn-primary\">Submit</button>");
+    out.println("</form>");
+
+  }
+
+  /** Returns the nickname of the user with id, or null if the user has not set a nickname. */
+  private String getUserNickname(String id) {
```

Comment:
this method is repeated.. Think about how this functionality could be organized so that both Servlets use the same code